ctapipe/instrument/subarray.py

```diff
@@ -314,6 +313,24 @@ def select_subarray(self, name, tel_ids):
         tel_positions = {tid: self.positions[tid] for tid in tel_ids}
         tel_descriptions = {tid: self.tel[tid] for tid in tel_ids}
 
+        if not name:
+            # sets don't interact nicely with numpy, so we transform it to a list
+            if isinstance(tel_ids, set):
+                tel_ids = list(tel_ids)
```

Comment:
just always do `tel_ids = sorted(tel_ids)`, that will work with any sequence and return a list with the sorted tel ids (which you rely on when using diff).